mbed_targets/mbed_targets.py

```diff
@@ -126,13 +120,27 @@ class MbedTargets(Set):
 
     @classmethod
     def from_offline_database(cls) -> "MbedTargets":
-        """Initialise with the offline target database."""
-        return cls(MbedTarget(**t) for t in target_database.get_offline_target_data())
+        """Initialise with the offline target database.
+
+        Raises:
+            TargetDataNotFound: Could not retrieve data from the target database.
+        """
+        try:
+            return cls(MbedTarget(**t) for t in target_database.get_offline_target_data())
+        except TargetDatabaseError as tgt_db_err:
+            raise TargetDataNotFound("Could not retrieve data from the target database.") from tgt_db_err
 
     @classmethod
     def from_online_database(cls) -> "MbedTargets":
-        """Initialise with the online target database."""
-        return cls(MbedTarget.from_target_entry(t) for t in target_database.get_online_target_data())
+        """Initialise with the online target database.
+
+        Raises:
+            TargetDataNotFound: Could not retrieve data from the target database.
+        """
+        try:
+            return cls(MbedTarget.from_target_entry(t) for t in target_database.get_online_target_data())
```

Comment:
FWIW, I think we should wrap only the code raising exception in `try/except` here. It's hard to figure out what can raise `TargetDatabaseError` by just reading this.

Reply:
We don't transform the exception anymore so this `try/except` has been removed.